core/src/main/java/com/amplifyframework/auth/options/AuthSignUpOptions.java

```diff
@@ -80,47 +75,73 @@ public boolean equals(Object obj) {
+     */
     @Override
     public String toString() {
         return "AuthSignUpOptions{" +
                 "userAttributes=" + userAttributes +
-                ", validationData=" + validationData +
                 '}';
     }
 
-    public static final class Builder {
-        private Map<String, String> userAttributes;
-        private Map<String, String> validationData;
+    public abstract static class Builder<T extends Builder<T>> {
+        private List<AuthUserAttribute> userAttributes;
 
+        /**
+         * Initialize the builder object with an empty array list for userAttributes.
+         */
         public Builder() {
-            this.userAttributes = new HashMap<>();
-            this.validationData = new HashMap<>();
+            this.userAttributes = new ArrayList<>();
         }
 
+        /**
+         * Return the type of builder this is so that chaining can work correctly without implicit casting.
+         * @return the type of builder this is
+         */
+        public abstract T getThis();
+
+        /**
+         * Returns the list of user attributes so extending classes can include this value in their constructor.
+         * @return the list of user attributes
+         */
+        protected List<AuthUserAttribute> getUserAttributes() {
+            return userAttributes;
+        }
+
+        /**
+         * List of user attribute keys and values to populate on sign up.
+         * @param userAttributes List of objects containing user attribute keys and values
+         * @return The type of Builder object being used.
+         */
         @NonNull
-        public Builder userAttributes(@NonNull Map<String, String> userAttributes) {
+        public T userAttributes(@NonNull List<AuthUserAttribute> userAttributes) {
             Objects.requireNonNull(userAttributes);
             this.userAttributes.clear();
-            this.userAttributes.putAll(userAttributes);
-            return this;
+            this.userAttributes.addAll(userAttributes);
+            return getThis();
         }
 
+        /**
+         * Build an instance of AuthSignUpOptions (or one of its subclasses).
+         * @return an instance of AuthSignUpOptions (or one of its subclasses)
+         */
         @NonNull
-        public Builder validationData(@NonNull Map<String, String> validationData) {
-            Objects.requireNonNull(validationData);
-            this.validationData.clear();
-            this.validationData.putAll(validationData);
-            return this;
+        public AuthSignUpOptions build() {
+            return new AuthSignUpOptions(Immutable.of(userAttributes));
         }
+    }
 
-        @NonNull
-        public AuthSignUpOptions build() {
-            return new AuthSignUpOptions(Immutable.of(userAttributes), Immutable.of(validationData));
+    public static final class CoreBuilder extends Builder<CoreBuilder> {
+
+        @Override
+        public CoreBuilder getThis() {
```

Comment:
Yeah I'd done a bunch of research online because Builder extension is actually quite a difficult problem and this seemed to be the most widely accepted answer to the problem.